### PR TITLE
iOS 16 Support

### PR DESCRIPTION
**Pull request by itsnebulalol**
This might need a bit more work, making a PR so I can merge with main.

- Does not mount user data partition for iPhone X compatibility, isn't even really needed anymore
- Deploys files to the rootfs (fakefs if needed)
- Fix deviceid finding
- Use apticket.der because dumping rdisk seems to freeze
- Add /.installed_palera1n with info
- uicache loader app on boot (no more Tips app hijacking)
- Fix rootless
- Semi-tethered webkit fix on 16
- Switch to local boot

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,8 +92,9 @@ If you want to make a rootless repo, use the official [palera1n repo](https://gi
...
 - [tihmstar](https://github.com/tihmstar) for [pzb](https://github.com/tihmstar/partialZipBrowser)/original [iBoot64Patcher](https://github.com/tihmstar/iBoot64Patcher)/original [liboffsetfinder64](https://github.com/tihmstar/liboffsetfinder64)/[img4tool](https://github.com/tihmstar/img4tool)
+- [Tom](https://github.com/guacaplushy) for local boot patch
 - [xerub](https://github.com/xerub) for [img4lib](https://github.com/xerub/img4lib) and [restored_external](https://github.com/xerub/sshrd) in the ramdisk
 - [Cryptic](https://github.com/Cryptiiiic) for [iBoot64Patcher](https://github.com/Cryptiiiic/iBoot64Patcher) fork, and [liboffsetfinder64](https://github.com/Cryptiiiic/liboffsetfinder64) fork
 - [libimobiledevice](https://github.com/libimobiledevice) for several tools used in this project (irecovery, ideviceenterrecovery etc), and [nikias](https://github.com/nikias) for keeping it up to date
-- [Nick Chan](https://github.com/asdfugil) general help with patches.
+- [Nick Chan](https://github.com/asdfugil) general help with patches and the base of the custom launchd/[haxx](https://github.com/asdfugil/haxx)
```
Comment (asdfugil) [inline README.md:99]: iBoot payload stuff

File: README.md [hunk 2]
```diff
@@ -93,8 +92,9 @@ If you want to make a rootless repo, use the official [palera1n repo](https://gi
...
 - [Sam Bingner](https://github.com/sbingner) for [Substitute](https://github.com/sbingner/substitute)
```
Comment (asdfugil) [inline README.md:100]: where dora2-iOS

File: other/rootfs/jbin/post.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+# link dyld if using semi tethered (iOS 16)
+if [ -d "/System/Cryptexes/OS/System/Library/Caches/com.apple.dyld" ]; then
+    if [ ! -d "/System/Library/Caches/com.apple.dyld" ]; then
+        $binpack/bin/ln -s /System/Cryptexes/OS/System/Library/Caches/com.apple.dyld /System/Library/Caches/
+    fi
+fi
+
+echo "[post.sh] done"
+exit
```
Comment (asdfugil) [inline other/rootfs/jbin/post.sh:17]: missing end of file new line
```suggestion
exit

```

File: palera1n.sh [hunk 4]
```diff
@@ -388,12 +374,18 @@ if [ -z "$tweaks" ] && [ "$semi_tethered" = "1" ]; then
     exit 1;
 fi
 
+if [ -z "$tweaks" ] && [ "$restorerootfs" = "1" ]; then
+    echo "[!] --restorerootfs may not be used with rootless"
+    echo "    You can click Remove in Pogo to get rid of the jailbreak"
+    exit 1;
```
Comment (asdfugil) [inline palera1n.sh:400]: how would kernelcachd be deleted

File: palera1n.sh [hunk 5]
```diff
@@ -388,12 +374,18 @@ if [ -z "$tweaks" ] && [ "$semi_tethered" = "1" ]; then
...
+fi
+
 if [ "$tweaks" = 1 ] && [ ! -e ".tweaksinstalled" ] && [ ! -e ".disclaimeragree" ] && [ -z "$semi_tethered" ] && [ -z "$restorerootfs" ]; then
     echo "!!! WARNING WARNING WARNING !!!"
     echo "This flag will add tweak support BUT WILL BE TETHERED."
     echo "THIS ALSO MEANS THAT YOU'LL NEED A PC EVERY TIME TO BOOT."
     echo "THIS ONLY WORKS ON 15.0-15.7.1"
```
Comment (asdfugil) [inline palera1n.sh:407]: 15.7.1 ???

File: palera1n.sh [hunk 6]
```diff
@@ -596,52 +610,40 @@ if [ ! -f blobs/"$deviceid"-"$version".shsh2 ]; then
...
-
     # lets actually patch the kernel
     echo "[*] Patching the kernel"
-    remote_cmd "rm -f /mnt6/$active/System/Library/Caches/com.apple.kernelcaches/kcache.raw /mnt6/$active/System/Library/Caches/com.apple.kernelcaches/kcache.patched /mnt6/$active/System/Library/Caches/com.apple.kernelcaches/kcache.im4p /mnt6/$active/System/Library/Caches/com.apple.kernelcaches/kernelcachd"
-    if [ "$semi_tethered" = "1" ]; then
-        remote_cmd "cp /mnt6/$active/System/Library/Caches/com.apple.kernelcaches/kernelcache /mnt6/$active/System/Library/Caches/com.apple.kernelcaches/kernelcache.bak"
+    remote_cmd "rm -f /mnt1/private/var/root/kpf"
+    if [[ "$version" == *"16"* ]]; then
+        if [ "$semi_tethered" = "1" ]; then
+            remote_cp binaries/Kernel16Patcher-nolivefs.ios root@localhost:/mnt1/private/var/root/kpf
```
Comment (asdfugil) [inline palera1n.sh:660]: I guess someone does not know how to parse arguments

**Review by asdfugil — CHANGES_REQUESTED** [review]

File: palera1n.sh (hunk 6 above)
Comment (DwifteJB) [reply palera1n.sh:660]: skull emoji

Comment (HideNames) [reply palera1n.sh:651]: Where can I find this file Kernel16Patcher-nolivefs.ios?
I need this file to try on iOS 16, thanks a lot

**Comment by dora2-iOS** [comment]
do you want to display the logo using setpicture?

File: palera1n.sh (hunk 6 above)
Comment (AriezzYT) [reply palera1n.sh]: on  god

**Pull request merged**